LiqoTech/liqo
# Added Advertisement status in PeeringRequest

# Description
This PR adds in the `PeeringRequest` object the information about Advertisement acceptance by the foreign cluster.
When the `Advertisement` is created on the foreign cluster, the broadcaster starts watching it: when the foreign cluster accepts/refuses the Advertisement (by setting the field `Status.AdvertisementStatus`), this information is copied in the local `PeeringRequest` object, allowing the home cluster to display it.

## Other changes
- added `AdvertisementStatus` field in PeeringRequest status
- added utility function to create a crdClient for PeeringRequest resources (following the same pattern used for other CRD)
- `WatchAdvertisement` function is now a method of the Broadcaster

# How Has This Been Tested?

- [x] unit test on the new feature:
1. create a `PeeringRequest` on home cluster
2. create an `Advertisement` on foreign cluster
3. set adv status to **ACCEPTED**
4. check the `PeeringRequest` status is set accordingly
